lib/read-json.js

```diff
@@ -348,11 +348,11 @@ function bins (file, data, cb) {
   data = normalizePackageBin(data)
 
   var m = data.directories && data.directories.bin
-  if (data.bin || !m) {
+  if (data.bin || !m || typeof m !== 'string') {
```

Comment:
Tests still pass without this change. What's it guarding for?

Before, if you set directories.bin.foo to an object, the path.resolve would throw. Probably not the most helpful message but it wouldn't allow the invalid non-string there.

Reply:
One of the big pushes we are making in npm 10 is to stop silently fixing bad entries in the package.json.

This is the module we use during publish so it's even more important to not introduce any new auto-correcting/ignoring behaviors.

Since it already throws an exception here, we should keep doing.  Having the thrown exception be more helpful would be nice, but not a deal breaker for this PR since it is technically out of scope.  It's your call.

Reply:
Reasonable. Reverted.